### ISSUE TITLE
port info

### ISSUE THREAD

**Issue by thewh1teagle** [de-vri-es/serial2-rs#22]
It will be very useful to get port info such as vid / hid name, etc...
so I can detect my device automatically.
Same as here:
https://github.com/serialport/serialport-rs/blob/819b64000e8c9758589443bc1af08e97a7bde2f2/src/lib.rs#L701

**Comment by de-vri-es** [comment]
I agree! This is related to #12 and should probably be solved with the same solution.

The main reason I haven't done it yet is the added complexity. But it would be a very useful feature.

I think we can not guarantee support for all platforms though. So I think all fields of the `SerialPortInfo` struct need to be options.

And we probably can't guarantee identical metadata across platforms for the same hardware either. But this can be solved with documentation.

**Comment by wc7086** [comment]
Currently, I am using https://crates.io/crates/serial_enumerator as a temporary solution.
